pkg/compute/management_client.go

```
@@ -0,0 +1,133 @@
+// it will periodically send an update to the requester node with
+// the latest node info for this node.
+type ManagementClient struct {
+	closeChannel      chan struct{}
+	labelsProvider    models.LabelsProvider
+	managementProxy   ManagementEndpoint
+	nodeID            string
+	nodeInfoDecorator models.NodeInfoDecorator
+	registrationFile  *RegistrationFile
+}
+
+func NewManagementClient(params ManagementClientParams) *ManagementClient {
+	return &ManagementClient{
+		closeChannel:      make(chan struct{}),
+		labelsProvider:    params.LabelsProvider,
+		managementProxy:   params.ManagementProxy,
+		nodeID:            params.NodeID,
+		nodeInfoDecorator: params.NodeInfoDecorator,
+		registrationFile:  NewRegistrationFile(params.NodeID),
+	}
+}
+
+func (m *ManagementClient) getNodeInfo(ctx context.Context) models.NodeInfo {
+	return m.nodeInfoDecorator.DecorateNodeInfo(ctx, models.NodeInfo{
+		NodeID:   m.nodeID,
+		NodeType: models.NodeTypeCompute,
+		Labels:   m.labelsProvider.GetLabels(ctx),
+	})
+}
+
+// RegisterNode sends a registration request to the requester node iff
+// we have not got evidence (a local sentinel file) that we have already
+// registered.  Should we fail to register, we will return an error and
+// expect the caller to exit.
+func (m *ManagementClient) RegisterNode(ctx context.Context) error {
+	// We only want to register this node if we haven't already
+	// been registered.
+	if m.registrationFile.Exists() {
+		log.Ctx(ctx).Debug().Msg("not registering with requester, already registered")
+		return nil
+	}
+
+	nodeInfo := m.getNodeInfo(ctx)
+	response, err := m.managementProxy.Register(ctx, requests.RegisterRequest{
+		Info: nodeInfo,
+	})
+	if err != nil {
+		return errors.New("failed to register with requester node")
+	}
+
+	if response.Accepted {
+		log.Ctx(ctx).Debug().Msg("register request accepted")
+		if err := m.registrationFile.Set(); err != nil {
+			log.Ctx(ctx).Error().Msgf("failed to record local registration status")
+		}
```

should we return an error here?